modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java

```diff
@@ -709,7 +684,20 @@ public ObjectDefinition putObjectDefinition(
 			objectValidationRules.toArray(new ObjectValidationRule[0]),
 			objectViews);
 
-		return _toObjectDefinition(serviceBuilderObjectDefinition);
+		if (status.getCode() != WorkflowConstants.STATUS_APPROVED) {
+			return _toObjectDefinition(serviceBuilderObjectDefinition);
+		}
+
+		serviceBuilderObjectDefinition =
```

Comment:
Why do we need to get the objectDefinition again from service?

Reply:
Actually taking a better look, we don't need to get again, I was trying to avoid an outdated object definition.

Done here: https://github.com/liferay-objects/liferay-portal/pull/3054/commits/aba956d5439f51997a93897de6e4b769e01d5dff